mycroft/configuration/config.py

```diff
@@ -268,3 +297,9 @@ def patch_clear(message):
         """
         Configuration.__patch = {}
         Configuration.load_config_stack(cache=True)
+
+    @staticmethod
+    def prune_config(config, prune_list):
+        for key in prune_list:
+            config = delete_key_from_dict(key, config)
+        return config
```

Comment:
Should this be a static method? I think this is mainly an internal thing and doesn't really need to be shipped with the Configuration object to the external caller. To me it can be a module level function. (also docstring)

Reply:
good call on both fronts :)